doc/common/Cadl-Project-Scripts.md

```diff
@@ -0,0 +1,127 @@
+# Integrating with cadl sync and generate scripts
+
+There are 2 common scripts provided for each language to be able to generate from within the language
+repo and use the remote cadl definition in the spec repo.
+
+## Prerequisites
+
+There are 4 things that these two scripts expect are set up in your language repo before they will run correctly.
+
+### cadl-location.yaml
+
+This file should live under the project directory for each service and has the following properties
+
+| Property | Description | IsRequired |
+| --- | --- | --- |
+| directory | The top level directory where the main.cadl for the serivce lives.  This should be relative to the spec repo root such as `specification/cognitiveservices/OpenAI.Inference` | true |
+| additionalDirectories | Sometimes a cadl file will use a relative import that might not be under the main directory.  In this case you can specify additional directories as a list to sync so that all needed files are synced. | false: default = null |
+| commit | The commit sha for the version of the cadl files you want to generate off of.  This allows us to have idempotence on generation until we opt into pointing at a later version. | true |
+| repo | The repo this spec lives in.  This should be either `Azure/azure-rest-api-specs` or `Azure/azure-rest-api-specs-pr`.  Note that pr will work locally but not in CI until we add another change to handle token based auth. | true |
+| cleanup | This will remove the TempCadlFiles directory after generation is complete if true otherwise this directory will be left to support local changes to the files to see how different changes would affect the generation. | false: default = true |
+
+Example
+
+```yml
+directory: specification/cognitiveservices/OpenAI.Inference
+additionalDirectories:
+  - specification/cognitiveservices/OpenAI.Authoring
+commit: 14f11cab735354c3e253045f7fbd2f1b9f90f7ca
+repo: Azure/azure-rest-api-specs
+cleanup: false
+```
+
+### TempCadlFiles
+
+You should add a new entry in your .gitignore for your repo so that none of these files are accidentally checked in if cleanup is turned off.
+`TempCadlFiles/`
```

Comment:
nit: we can instead give the gitignore command to give to people
```md
# .gitignore file
*TempCadlFiles/
```

Reply:
Thats what I did other than the comment, the * isn't needed.  That just means if I had another directory named `XTempCadlFiles` it would also be ignored.

Should I add the comment so its more obvious?

Reply:
Since this is a one time operation, as opposed to the cadl-location.yaml that is a one per package operation (will happen often), I believe we should clarify that in a section do not show too obviously and take place in a doc like that people will refer too often